tasks/process-spectrum.js

```diff
@@ -627,30 +627,17 @@ async function processComponent(componentPath) {
                             }
                         },
                     },
-                    filename: systemsPath,
+                    filename: overridesPath,
                 });
-
-                // if the code is an empty buffer then don't write the file
-                if (code.length != 1) {
-                    fs.writeFileSync(
-                        systemsPath,
-                        `/*
-    Copyright 2023 Adobe. All rights reserved.
-    This file is licensed to you under the Apache License, Version 2.0 (the "License");
-    you may not use this file except in compliance with the License. You may obtain a copy
-    of the License at http://www.apache.org/licenses/LICENSE-2.0
-    
-    Unless required by applicable law or agreed to in writing, software distributed under
-    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
-    OF ANY KIND, either express or implied. See the License for the specific language
-    governing permissions and limitations under the License.
-    */
-    
-    /* THIS FILE IS MACHINE GENERATED. DO NOT EDIT */
-            ${code}
-            `.replace(/\/\*\![\w|\W]*\*\//, '')
-                    );
-                }
+                // Note: We write the overrides file even if it's empty.
+                // This is to ensure that we don't end up with stale overrides
+                // files in the case where the bridge file previously contained
+                // overrides but no longer does.
+                writeMachineGeneratedSourceFile(overridesPath, code);
+            } else {
+                // For the same reason, we write an empty file if the bridge file
+                // doesn't exist (in case it previously did).
+                writeMachineGeneratedSourceFile(overridesPath, '');
```

Comment:
Should we delete the file entirely if it’s empty instead of writing an empty file? 💭

Reply:
Good question! That was my first instinct as well, but then I recalled that these generated `foo-override.css` files are imported by manually maintained `foo.css` files.

If we allow the override files to be routinely created / deleted, we would either need to start having the script modify the currently-manually-maintained files to update the imports, OR have the script output some list of deleted override files and depend on ourselves to make the corresponding edits.

Neither of those solutions seemed great, and since most of the components do in fact have overrides files, it seemed like unconditionally creating the overrides was the least-bad option. I'm open to being convinced otherwise, though!

Reply:
Of course, I'm hoping that all of this stuff can be made to go away in our post-1.x setup!